src/components/DiffView/index.tsx

```diff
@@ -111,4 +138,7 @@ class DiffView extends React.Component<PublicProps> {
   }
 }
 
-export default DiffView;
+// TODO: I can't make it work, I don't understand what's going on between
+// public, default and router props. TS gets confused by `withRouter` HOC.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export default withRouter(DiffViewBase) as any;
```

Comment:
Yes, I will give it another try to see if I can do anything, but this diff sounds like a better hack in the meantime.

Reply:
I think I understand what's going on now. The `withRouter` type is a bit weird but it does make sense: it takes a type `P` that extends the router props type and omits it when it returns the component so that the result of `withRouter<P>()` is of type `P`.

That makes sense but TS is confused because if `P` contains other props, e.g., from Redux, it can't tell the difference between required props from `PublicProps` and already provided props from Redux, hence the huge error with `InstrinsicAttributes`.

A better fix would be to be more descriptive about what the public API (i.e. public type) of our component is. I believe it is `PublicProps & Partial<DefaultProps>` so that we prevent rendering the component with unknown props and we also allow dependency injection in the test file.

Reply:
I think I understand what's going on now. The `withRouter` type is a bit weird but it does make sense: it takes a type `P` that extends the router props type and omits it when it returns the component so that the result of `withRouter<P>()` is of type `P`.

That makes sense but TS is confused because if `P` contains other props, e.g., from Redux, it can't tell the difference between required props from `PublicProps` and already provided props from Redux, hence the huge error with `InstrinsicAttributes`. This is not exactly what's happening here but I found a similar issue in another component. I believe here, TS gets confused because it looses track of the default props (TS support for that is very new).

A better fix would be to be more descriptive about what the public API (i.e. public type) of our component is. I believe it is `PublicProps & Partial<DefaultProps>` so that we prevent rendering the component with unknown props and we also allow dependency injection in the test file.

Reply:
I think I understand what's going on now. The `withRouter` type is a bit weird but it does make sense: it takes a type `P` that extends the router props type and omits it when it returns the component so that the result of `withRouter<P>()` is of type `P`.

That makes sense but TS is confused because if `P` contains other props, e.g., from Redux, it can't tell the difference between required props from `PublicProps` and already provided props from Redux, hence the huge error with `InstrinsicAttributes`. This is not exactly what's happening here but I found a similar issue in another component. I believe here, TS gets confused because it looses track of the default props (TS support for that is very new).

A better fix would be to be more descriptive about what the public API (i.e. public type) of our component is. I believe it is `PublicProps & Partial<DefaultProps>` so that we prevent rendering the component with unknown props and we also allow dependency injection in the test file.